### DISCUSSION TITLE
C++ level

### DISCUSSION THREAD

**Discussion by ThirtySomething** [SourceMonitor/SM-Info#3] [General]
Which C++ level should be used for the further SourceMonitor development?

**Comment by CharleO** [comment]
I've used SourceMonitor for quite a few years and still use today. My work is 95% embedded so we are limited to the tool manufacture slowly adding C++ library features. I would start with C++2011, as kick-off. So I use "https://en.cppreference.com/" to determine what the manufacture integrates into their tool, which I copy specific C++ library feature.

**Reply by ThirtySomething** [reply]
Hello @CharleO ,

thanks for your proposal. This make me thinking because we have at least two different C++ levels. This discussion was aimed for the SourceMonitor development itself. As far as I understood your post, you aimed about the C++ level SourceMonitor should process. These are two different things.

At least it looks like we'll use and maybe extend the grammars provided by the [ANTLR4 project](https://github.com/antlr/grammars-v4). But this is far away in the future. First we need to get SourceMonitor to become open source, which is at least a big bunch of work.
